src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java

```diff
@@ -12,44 +13,71 @@
 public class TicTacToeGame {
 
     private Board board;
+    private Scoreboard scoreboard;
+    private boolean gameFinished;
 
-    public TicTacToeGame(){
+    public TicTacToeGame() {
         board = new Board();
+        scoreboard = new Scoreboard();
+        gameFinished = false;
     }
 
-    public void promptNextPlayer(){
-        switch(board.getCurrentPlayer()){
+    public void promptNextPlayer() {
+        switch (board.getCurrentPlayer()) {
             case X:
                 System.out.println("It's player " + board.getSymbol(board.getCurrentPlayer()) + "'s turn. Please enter the coordinates of your next move as x,y: ");
                 break;
             case O:
                 System.out.println("It's player " + board.getSymbol(board.getCurrentPlayer()) + "'s turn. Please enter the coordinates of your next move as x,y: ");
                 break;
-
         }
     }
 
-    public void playGame(){
+    public void playGame() {
         Scanner keyboardScanner = new Scanner(System.in);
 
-        while (board.getWinner() == null){
+        while (!gameFinished) {
+            while (board.getWinner() == null && !board.getIsTie()) {
+                board.printBoard();
+                promptNextPlayer();
+                String line = keyboardScanner.nextLine();
+                String input[] = line.split(",");
+                try {
+                    board.playMove(Integer.parseInt(input[0]), Integer.parseInt(input[1]));
+                } catch (InvalidMoveException e) {
+                    System.out.println("Invalid coordinates. Try again");
+                    promptNextPlayer();
+                }
+            }
+
             board.printBoard();
-            promptNextPlayer();
+
+            if (board.getIsTie()) {
+                scoreboard.setTie(scoreboard.getTie() + 1);
+                System.out.println("It is a tie!");
+            } else {
+                if (board.getWinner().equals(Player.X)) {
+                    scoreboard.setPlayerXScore(scoreboard.getPlayerXScore() + 1);
+                } else {
+                    scoreboard.setPlayerOScore(scoreboard.getPlayerOScore() + 1);
+                }
+
+                System.out.println("Player " + board.getWinner() + " has won the game!");
+            }
+
+            scoreboard.showScoreboard();
```

Comment:
Should the scoreboard be shown after each win/tie or just at the end when the game is finished?

Reply:
Based on the assignment it should be when the game is finished, but it was asked in the eclass, and based on that we need to display it after each win/tie